src/ngraph/builder/quantization/quantized_linear_matmul.cpp

```diff
@@ -94,6 +94,54 @@ namespace ngraph
                 auto output_scale = make_constant(element::f32, Shape{}, 1);
                 return make_shared<op::QuantizedDot>(input0, input1, output_scale, false, false);
             }
+
+            shared_ptr<Node>
+                QuantizedLinearMatmulInteger(const std::shared_ptr<Node>& input0,
+                                             const std::shared_ptr<Node>& input1,
+                                             const std::shared_ptr<Node>& input0_zero_point,
+                                             const std::shared_ptr<Node>& input1_zero_point)
+            {
+                auto input0_zero = dynamic_pointer_cast<ngraph::op::Constant>(input0_zero_point);
+                auto input1_zero = dynamic_pointer_cast<ngraph::op::Constant>(input1_zero_point);
+
+                // Check if zero points are constant and zero
+                if (input0_zero != nullptr && input1_zero != nullptr &&
+                    ngraph::is_zero(input0_zero) && ngraph::is_zero(input1_zero))
+                {
+                    return QuantizedLinearMatmulInteger(input0, input1);
+                }
+                else
+                {
+                    // Fall back to performing matmul on dequantized floating-point values
+                    auto input0_scale = make_constant(element::f32, Shape{}, 1);
+                    auto input1_scale = make_constant(element::f32, Shape{}, 1);
+                    auto output_scale = make_constant(element::f32, Shape{}, 1);
+                    auto output_zero_point = make_constant(element::i32, Shape{}, 0);
+                    AxisSet axes;
+
+                    auto dq_input0 = make_shared<op::Dequantize>(input0,
+                                                                 input0_scale,
+                                                                 input0_zero_point,
+                                                                 input0_scale->get_element_type(),
+                                                                 axes);
+
+                    auto dq_input1 = make_shared<op::Dequantize>(input1,
+                                                                 input1_scale,
+                                                                 input1_zero_point,
+                                                                 input1_scale->get_element_type(),
+                                                                 axes);
+
+                    auto dot = make_shared<op::Dot>(dq_input0, dq_input1, 1);
+                    auto q_dot = make_shared<op::Quantize>(
+                        dot,
+                        output_scale,
+                        output_zero_point,
+                        output_zero_point->get_element_type(),
+                        axes,
+                        op::Quantize::RoundMode::ROUND_NEAREST_TOWARD_EVEN);
+                    return move(q_dot);
```

Comment:
In general you should avoid moving the return value, let the compiler choose the best way to optimize it. You can return the `make_shared` result directly in here.

Reply:
Without `move` this raises a compiler warning: 
```
warning: prior to the resolution of a defect report against ISO C++11, 
    local variable 'q_dot' would have been copied despite being returned 
    by name, due to its not matching the function return type```

Reply:
Done.